config/samples/pipeline_v1beta1_pipeline.yaml

```diff
@@ -7,46 +7,60 @@ spec:
     tasks:
         - name: 'test'
           taskRef:
-              name: test-wizzbang-task
-          sourceBindings:
-              - inputName: 'repoUnderTest'
-                sourceKey: 'wizzbang'
-          nextTasks: ['buildPush']
+            name: test-wizzbang-task
+          inputsourceBindings:
+            - name: repoUnderTest
+              sourceKey: wizzbang
         - name: 'buildPush'
           taskRef:
-              name: build-push-task
-          sourceBindings:
-              - inputName: 'workspace'
-                sourceKey: 'wizzbang'
-          artifactStoreBindings:
-              - storeName: 'registry'
-                storeKey: 'stagingRegistry'
+            name: build-push-task
+          inputSourceBindings:
+            - name: 'workspace'
+              sourceKey: 'wizzbang'
+              passedConstraint: [test-wizzbang-task]
+          outputSourceBindings:
+            - name: 'registry'
+              sourceKey: 'stagingRegistry'
           params:
-              - name: 'pathToDockerfile'
-                value: 'build/Dockerfile'
+            - name: 'pathToDockerfile'
+              type: 'build/Dockerfile'
         - name: 'integrationTest'
           taskRef:
-              name: integration-test-wizzbang-task
-          sourceBindings:
-              - inputName: 'repoUnderTest'
-                sourceKey: 'wizzbang'
-          paramBindings:
-              - inputName: testImage
-                taskName: buildPush
-                taskOutputName: builtImage
-        - name: 'deploy'
+            name: integration-test-wizzbang-task
+          inputSourceBindings:
+            - name: repoUnderTest
+              sourceKey: wizzbang
+              passedConstraint: [test]
+          outputSourceBindings:
+            - name: stagingRegistry
+              sourceKey: stagingRegistry
+              passedConstraint: [build-push-task]
```

Comment:
to fully understand the new design, I think the Tasks these are referring to need to be updated as well, for example in [integration-test-wizzbang-task](https://github.com/knative/build-pipeline/blob/master/examples/test_tasks.yaml) right now we've got:

```yaml
    inputs:
        sources:
            - name: workspace
        params:
            - name: testImage
              type: string
            - name: testArgs
              type: string
    outputs:
        results:
            - name: testResults
              format: junit
              path: integration/logs/tests.xml
```

But it looks like the `outputs` should be updated to indicate that the Task will be requiring an Output resource of type `image`?

Reply:
we updated some of them, must have missed this one, will update. You are correct in that the task needs an output resource of image if it is creating this image and passing it along.